flash-patcher.py

```diff
@@ -220,293 +361,455 @@ def apply_patch(patch_file):
 
-    if not os.path.exists(inputfile):
-        perror("Could not locate the SWF file: " + inputfile)
-        perror("Aborting...")
-        exit(1)
 
-    cache_location = "./.Patcher-Temp/" + base64.b32encode(bytes(inputfile, "utf-8")).decode("ascii")
+def decompile_swf(
+    iface: JPEXSInterface,
+    inputfile: str,
+    *,
+    drop_cache: bool = False,
+    xml_mode: bool = False,
+) -> Path:
+    """Decompile the SWF and return the decompilation location.
+
+    This uses caching to save time.
+
+    inputfile: the SWF to decompile
+    drop_cache: if True, will force decompilation instead of using cached files
+    """
+    # Decompile swf into temp folder called ./.Patcher-Temp/[swf name, base32 encoded]
+    if not Path("./.Patcher-Temp").exists():
+        Path("./.Patcher-Temp").mkdir()
+
+    if not inputfile.exists():
+        error(
+            """Could not locate the SWF file: %s.
+            Aborting...""",
+            inputfile,
+        )
+        sys.exit(1)
+
+    cache_location = Path(
+        "./.Patcher-Temp/",
+        base64.b32encode(
+            bytes(inputfile.name, "utf-8"),
+        ).decode("ascii"),
+    )
+
     if xml_mode:
+        info("XML decompilation mode.")
         global DECOMP_LOCATION
         global DECOMP_LOCATION_WITH_SCRIPTS
 
-        cache_location = "./.Patcher-Temp/swf2.xml"
+        cache_location = Path("./.Patcher-Temp/swf2.xml")
 
-        DECOMP_LOCATION = "./.Patcher-Temp/swf.xml"
-        DECOMP_LOCATION_WITH_SCRIPTS = "./.Patcher-Temp/"
+        DECOMP_LOCATION = Path("./.Patcher-Temp/swf.xml")
+        DECOMP_LOCATION_WITH_SCRIPTS = Path("./.Patcher-Temp/")
 
     # Mkdir / check for cache
-    if invalidate_cache or (not os.path.exists(cache_location)):
-        if (not os.path.exists(cache_location) and not xml_mode):
-            os.mkdir(cache_location)
+    if drop_cache or (not Path(cache_location).exists()):
+        if not Path(cache_location).exists() and not xml_mode:
+            Path(cache_location).mkdir()
 
-        print("Beginning decompilation...")
+        info("Beginning decompilation...")
 
         decomp = None
 
         if xml_mode:
-            decomp = subprocess.run([JPEXS_PATH] + JPEXS_ARGS + ["-swf2xml", inputfile, cache_location], \
-                stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
-            
+            decomp = iface.dump_xml(inputfile, cache_location)
+            info("XML decompilation mode.")
         else:
-            decomp = subprocess.run([JPEXS_PATH] + JPEXS_ARGS + ["-export", "script", cache_location, inputfile], \
-                stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
-        
-        if (decomp.returncode != 0):
-            perror("JPEXS was unable to decompile the SWF file: " + inputfile)
-            perror("Aborting...")
-            exit(1)
+            decomp = iface.export_scripts(inputfile, cache_location)
+
+        if decomp.returncode != 0:
+            error(
```

Comment:
nit: error -> exception

Reply:
Can't use `exception()` outside of `except` blocks: https://docs.python.org/3/library/logging.html#logging.exception